sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpRuleManager.cs

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.ServiceBus.Amqp.Framing;
+using Azure.Messaging.ServiceBus.Core;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Encoding;
+
+namespace Azure.Messaging.ServiceBus.Amqp
+{
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable. AmqpRuleManager does not own connection scope.
+    internal class AmqpRuleManager : TransportRuleManager
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+    {
+        /// <summary>
+        /// The path of the Service Bus subscription to which the rule manager is bound.
+        /// </summary>
+        ///
+        private readonly string _subscriptionPath;
+
+        /// <summary>
+        /// The policy to use for determining retry behavior for when an operation fails.
+        /// </summary>
+        private readonly ServiceBusRetryPolicy _retryPolicy;
+
+        /// <summary>
+        /// The identifier for the rule manager.
+        /// </summary>
+        private readonly string _identifier;
+
+        /// <summary>
+        /// The AMQP connection scope responsible for managing transport constructs for this instance.
+        /// </summary>
+        ///
+        private readonly AmqpConnectionScope _connectionScope;
+
+        /// <summary>
+        /// Indicates whether or not this instance has been closed.
+        /// </summary>
+        private bool _closed;
+
+        /// <summary>
+        /// Indicates whether or not this rule manager has been closed.
+        /// </summary>
+        ///
+        /// <value>
+        /// <c>true</c> if the rule manager is closed; otherwise, <c>false</c>.
+        /// </value>
+        public override bool IsClosed => _closed;
+
+        private readonly FaultTolerantAmqpObject<RequestResponseAmqpLink> _managementLink;
+
+        static AmqpRuleManager()
+        {
+            AmqpCodec.RegisterKnownTypes(AmqpTrueRuleFilterCodec.Name, AmqpTrueRuleFilterCodec.Code, () => new AmqpTrueRuleFilterCodec());
```

Comment:
Interesting.   Never seen this before.   I'd be very interested in learning the details at some point, if you have time.